src/Middleware/PermissionMiddleware.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Spatie\Permission\Middleware;
+
+use Auth;
+use Closure;
+
+class PermissionMiddleware
+{
+    public function handle($request, Closure $next, $permission)
+    {
+        if (Auth::guest()) {
+            return $next($request);
```

Comment:
@freekmurze I need resolve this  line  ```return redirect($urlOfYourLoginPage);```
My first thought was create a config element to replace $urlOfYourLoginPage. But then why this work not would correspond to middelware Auth?

But when you say this, I understand your position. 

I return my initial idea, add config params to define url to redirect. Perhaps ... ?

```php
if (Auth::guest()) {
    return redirect(config('permission.url_redirect_guest', '/login'));
```